### ISSUE TITLE
Loan Section feature

### ISSUE THREAD

**Issue by arnavarya89** [Ivy-Apps/ivy-wallet#2740]
### Please confirm the following

- [x] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.

### What would you like to improve?

In the Loan Option Suppose I lend a loan or money to a person with Reason X then I again gave money to the same person with reason Y and many times. So instead of adding a loan many times can't we do that in the add record button we again add some money as lent with both reason X and Y visible on it will be more clean to see it and easy to maintain records and for the borrowed part

### Because...

It would be easier for all users to maintain their records more conveniently 

### Success Criteria

_add a option in add record option to like increase debit with a reason for loan  just a same as the features to receive money we just to do it with increase debt.

**Comment by arnavarya89** [comment]
Ok no issue but can you resolve my query
Suppose I gave some money to a person first for some reason A and then
again I gave money for reason B then he returns me some money and more then
how do I maintain the records of these.
Please help me with this

**Comment by arnavarya89** [comment]
Yep now you got the problem. But I didn't get your solution can you explain
in detail.

**Comment by vartult** [comment]
Hey @nvllz, correct me if I'm wrong, we can easily update the total borrowed amount by updating the master borrowed amount, like in the example you took we can basically update the amount to 290USD. Though this won't create an entry for the txn but acknowledge the usecase OP wants. We can then track owed money repayment by adding records. Though we can give an option while updating the master borrowed amount to create a separate entry for this amount update.

**Comment by arnavarya89** [comment]
Yes, Suppose I increase debt for a reason then after some days I will
forgot the reason then it might be debatable with the person why I
increases debt in repayment of money.

**Comment by vartult** [comment]
Sure, @arnavarya89 do you have a solution to this? or I can brainstorm around it and impl.

**Comment by arnavarya89** [comment]
Sure I have an solution to it in fact I have two solution to it first one
is let me explain you by an example to make you understand more better,
In the Loan Option Suppose I lend a loan or money to a person with Reason X
then I again gave money to the same person with reason Y and many times. So
instead of adding a loan many times can't we do that in the add record
button we again add some money as lent with both reason X and Y visible on
it will be more clean to see it and easy to maintain records.

And the second one is in records button to add interest option just you
need to add a description option and increase or decrease debt accordingly
and not in the add interest.

Summarising both solution I would say that for the first solution instead
of add loan for same person different time it should be add record so that
we can increase or decrease loan with a reason in one type and not
different times for same person not in different loan.
And for the second you Just need to add description option in intrest
option and I recommend you that rename it to increase debt and it should be
shown in main loan amount not in intrest amount If you would ask me then if
you say then I would attach other app debt options how they implemented it
in their app so you would get a clear pictures of it.

**Comment by arnavarya89** [comment]
No response yet ??

**Comment by ILIYANGERMANOV** [comment]
It's unassigned if anyone wants to take it

**Comment by ILIYANGERMANOV** [comment]
> I was about to place a feature request for this, but someone else already has that problem. Is there a way to upvote this or something to make the devs know that other users are also interested?

Hey @RDjarbeng we don't have an upvote mechanism - maybe just react with 👍

**Comment by RDjarbeng** [comment]
Alright, noted. Looking forward to this feature. Thank you for building this app too

**Comment by michalguspiel** [comment]
I'm on it

**Comment by ivywallet** [comment]
Thank you for your interest @michalguspiel! 🎉
Issue #2740 is assigned to you. You can work on it! ✅

_If you don't want to work on it now, please un-assign yourself so other contributors can take it._

Also, make sure to read our [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md).

**Comment by RDjarbeng** [comment]
Thank you so much. Will this be available on the Google Play Store now or we have to wait for an update?
